tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -433,6 +436,57 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
 		framework.ExpectNoError(err)
 		// TODO: Add pump configmap rolling-update case
 	})
+
+	ginkgo.It("API: Migrate from helm to CRD", func() {
+		cluster := newTidbClusterConfig(e2econfig.TestConfig, ns, "helm-migration", "admin", "")
+		cluster.Resources["pd.replicas"] = "1"
+		cluster.Resources["tikv.replicas"] = "1"
+		cluster.Resources["tidb.replicas"] = "1"
+		oa.DeployTidbClusterOrDie(&cluster)
+		oa.CheckTidbClusterStatusOrDie(&cluster)
+
+		tc, err := cli.PingcapV1alpha1().TidbClusters(cluster.Namespace).Get(cluster.ClusterName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "Expected get tidbcluster")
+
+		// TODO: modify other cases to manage TiDB configmap in CRD by default
+		ginkgo.By("Test managing TiDB configmap in TidbCluster CRD")
+		tc.Spec.TiDB.Config = &v1alpha1.TiDBConfig{}
+		tc.Spec.TiDB.ConfigUpdateStrategy = v1alpha1.ConfigUpdateStrategyInPlace
+
+		_, err = cli.PingcapV1alpha1().TidbClusters(tc.Namespace).Update(tc)
+		framework.ExpectNoError(err, "Expected update tidbcluster")
+
+		tidbSetName := controller.TiDBMemberName(tc.Name)
+		oldTiDBSet, err := c.AppsV1().StatefulSets(tc.Namespace).Get(tidbSetName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "Expected get TiDB statefulset")
```

Comment:
should we get old TiDB statefulset before updating TiDB cluster?

Reply:
Yes, this case has to verify the controllerrevision will not be changed after we manage configmap in controller (set `.spec.tidb.config` to non-nil)

Reply:
this is what I think:
- deploy tidb cluster
- wait tidb cluster is fully up
- get tidb statefulset
- update tidb cluster
- check if tidb cluster is not upgraded in a few minutes (e.g.  if the tidb cluster is not upgraded in 2 minutes, then we think tidb-operator will not upgrade tidb cluster because of our change)
- wait for our check is ok in 5 minutes
  - compare with old tidb set before updating tidb cluster

Reply:
Yes, wait for a certain period is a safer choice. I would check the tidb-servers are not upgrade for 1 minutes after the tidb-statefulset get adopted by `tidbcluster`, because it may takes time for `tidb-controller-manager` to sync the statefulset, and after the statefulset is fully synced, 1 minute should be enough for `kube-controller-manager` to determine whether the statefulset should be rolling-udpated

Reply:
oh, tidb-controller do not adopt statefulset, my fault